avm/res/synapse/workspace/main.bicep

```diff
@@ -0,0 +1,483 @@
+param azureADOnlyAuthentication bool = false
+
+@description('Optional. AAD object ID of initial workspace admin.')
+param initialWorkspaceAdminObjectID string = ''
+
+@description('Required. Resource ID of the default ADLS Gen2 storage account.')
+param defaultDataLakeStorageAccountResourceId string
+
+@description('Required. The default ADLS Gen2 file system.')
+param defaultDataLakeStorageFilesystem string
+
+@description('Optional. Create managed private endpoint to the default storage account or not. If Yes is selected, a managed private endpoint connection request is sent to the workspace\'s primary Data Lake Storage Gen2 account for Spark pools to access data. This must be approved by an owner of the storage account.')
+param defaultDataLakeStorageCreateManagedPrivateEndpoint bool = false
+
+@description('Optional. The customer managed key definition.')
+param customerManagedKey customerManagedKeyType
+
+@description('Optional. Activate workspace by adding the system managed identity in the KeyVault containing the customer managed key and activating the workspace.')
+param encryptionActivateWorkspace bool = false
+
+@maxLength(90)
+@description('Optional. Workspace managed resource group. The resource group name uniquely identifies the resource group within the user subscriptionId. The resource group name must be no longer than 90 characters long, and must be alphanumeric characters (Char.IsLetterOrDigit()) and \'-\', \'_\', \'(\', \')\' and\'.\'. Note that the name cannot end with \'.\'.')
+param managedResourceGroupName string = ''
+
+@description('Optional. Enable this to ensure that connection from your workspace to your data sources use Azure Private Links. You can create managed private endpoints to your data sources.')
+param managedVirtualNetwork bool = false
+
+@description('Optional. The Integration Runtimes to create.')
+param integrationRuntimes array = []
+
+@description('Optional. Allowed AAD Tenant IDs For Linking.')
+param allowedAadTenantIdsForLinking array = []
+
+@description('Optional. Linked Access Check On Target Resource.')
+param linkedAccessCheckOnTargetResource bool = false
+
+@description('Optional. Prevent Data Exfiltration.')
+param preventDataExfiltration bool = false
+
+@allowed([
+  'Enabled'
+  'Disabled'
+])
+@description('Optional. Enable or Disable public network access to workspace.')
+param publicNetworkAccess string = 'Enabled'
+
+@description('Optional. Purview Resource ID.')
+param purviewResourceID string = ''
+
+@description('Required. Login for administrator access to the workspace\'s SQL pools.')
+param sqlAdministratorLogin string
+
+@description('Optional. Password for administrator access to the workspace\'s SQL pools. If you don\'t provide a password, one will be automatically generated. You can change the password later.')
+#disable-next-line secure-secrets-in-params // Not a secret
+param sqlAdministratorLoginPassword string = ''
```

Comment:
Are we sure this isn't a secret? It it a secret resourceId?

Reply:
I agree it should be considered a secret. I've updated the code with @secret(). Also updated test files with resourceLocation instead of location